x/pool-incentives/keeper/keeper.go

```diff
@@ -132,7 +138,26 @@ func (k Keeper) SetPoolGaugeId(ctx sdk.Context, poolId uint64, lockableDuration
 	store.Set(key, sdk.Uint64ToBigEndian(poolId))
 }
 
+func (k Keeper) SetPoolGaugeIdNoLock(ctx sdk.Context, poolId uint64, gaugeId uint64) {
+	store := ctx.KVStore(k.storeKey)
+	// maps pool id and gauge id to gauge id.
+	// Note: this could be pool id and gauge id to empty byte array,
+	// but is chosen this way for ease of implementation at the cost of space.
+	key := types.GetPoolNoLockGaugeIdStoreKey(poolId, gaugeId)
+	store.Set(key, sdk.Uint64ToBigEndian(gaugeId))
+
+	key = types.GetPoolIdFromGaugeIdStoreKey(gaugeId, 0)
+	store.Set(key, sdk.Uint64ToBigEndian(poolId))
+}
+
+// GetPoolGaugeId returns the gauge id associated with the pool id and lockable duration.
+// This can only be used for the internally incentivized gauges.
+// Externally incentivized gauges do not have such link created.
 func (k Keeper) GetPoolGaugeId(ctx sdk.Context, poolId uint64, lockableDuration time.Duration) (uint64, error) {
```

Comment:
aah so now for queries, this will return all internal gauges? and `GetNoLockGaugeIdsFromPool` will return all external gauges?

Reply:
Also i genuinely thought this was a query specific problem. Can we not resolve it(retrieve based on internal/external denoms) at query level i.e. `QueryIncentivizedPools` and `QueryExternalIncentiveGaugesRequest`

Reply:
`GetNoLockGaugeIdsFromPool` returns **all** "NoLock" gauges associated with a pool id.

`GetPoolGaugeId` returns any **internal** gauge associated with a pool id and a local duration.
For "ByDuration" gauges, this requires a lockableDuration to be specified as there can be 3 duratons AFAIK.
For "NoLock"" gauges, this requires incentivized epoch duration to be specified as that's what we set it to when creating a pool.

Reply:
that sounds good